### PR TITLE
preload next video (#82)

### PR DESCRIPTION
**Pull request by goatandsheep**
## Description

This feature preloads the next video at the beginning of the previous video to ensure time between videos is closer to the specified delay.

https://github.com/brightcove/videojs-playlist/issues/82

## Specific Changes proposed

Replaced the autoadvance timeout function with a function that preloads the next video.

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [x] If necessary, more likely in a feature request than a bug fix
  - [ ] Unit Tests updated or fixed
  - [ ] Docs/guides updated
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

**Comment by goatandsheep** [comment]
It doesn't handle HLS, yet

File: src/auto-advance.js [hunk 1]
```diff
@@ -67,12 +67,27 @@ const setup = (player, delay) => {
     // we need to cancel the auto-advance. This could mean the user seeked
     // back into the content or restarted the content. This is reproducible
     // with an auto-advance > 0.
-    player.one('play', cancelOnPlay);
+    // player.one('play', cancelOnPlay);
```
Comment (goatandsheep) [inline src/auto-advance.js]: can someone help me understand the errors? I'm assuming it's because I commented this out

**Comment by cerebroo** [comment]
Not working on chrome as link preload as video is not supported by chrome anymore.

**Comment by goatandsheep** [comment]
I'm not sure if I understand as there is a [video preload tutorial](https://developers.google.com/web/fundamentals/media/fast-playback-with-video-preload) and [preload is supported by chrome](https://caniuse.com/#search=preload)

**Comment by cerebroo** [comment]
@goatandsheep - Yes preload is supported by Chrome but not `as="video"`. The tutorial you linked to has a method `preloadFullVideoSupported`. I tried that in chrome 74.0.3729.131 on Mac and it returns false.

Infact it gives following error on chrome.
<link rel=preload> must have a valid `as` value

Also answer here suggest that Chrome has removed preload as video/audio support.
https://stackoverflow.com/questions/50593084/link-rel-preload-must-have-a-valid-as-value

**Comment by goatandsheep** [comment]
What about `as="object"`?

**Comment by Moongazer** [comment]
Are there any news on this? I'm trying to build a playlist of short video snippets with seamless playback, but whatever I do, there is always a ~500ms gap in between (even if the video files coming locally from my dev server)

**Comment by tulkas85** [comment]
any news ? I try to add in autoadvance timeout this but doesn'works 
```

const n = player.playlist.nextIndex();

      if (n >= 0) {
        // determine source
        const preloadlocation = player.playlist()[n].sources[0].src;
        const preloadLink = document.createElement('link');

        preloadLink.href = preloadlocation;
        preloadLink.crossOrigin = 'anonymous';
        preloadLink.rel = 'preload';
        preloadLink.as = 'object';
        document.head.appendChild(preloadLink);
      }
```

**Comment by goatandsheep** [comment]
I think this project is abandoned

**Comment by goatandsheep** [comment]
Hi @misteroneill I know this pr is likely completely out of date with head. Is this feature actually implemented in this project?

**Pull request closed without merging**